book-content/chapters/04-essential-types-and-annotations.md

```diff
@@ -0,0 +1,2366 @@
+
+So, let's update the function declaration parameters so that `a` and `b` are both specified as `string`:
+
+```ts twoslash
+const concatTwoStrings = (a: string, b: string) => {
+  return [a, b].join(" ");
+};
+```
+
+This change fixes the errors.
+
+For a bonus point, what type will the return type be inferred as?
+
+```ts twoslash
+const concatTwoStrings = (a: string, b: string) => {
+  return [a, b].join(" ");
+};
+
+// ---cut---
+const result = concatTwoStrings("Hello", "World");
+//    ^?
+```
+
+#### Solution 3: Updating Basic Types
+
+Each of the examples represents the TypeScript's basic types, and would be annotated as follows:
+
+```ts twoslash
+let example1: string = "Hello World!";
+let example2: number = 42;
+let example3: boolean = true;
+let example4: symbol = Symbol();
+let example5: bigint = 123n;
+```
+
+We've already seen `string`, `number`, and `boolean`. The `symbol` type is used for `Symbol`s which are used to ensure property keys are unique. The `bigint` type is used for numbers that are too large for the `number` type.
+
+However, in practice you mostly won't annotate variables like this. If we remove the explicit type annotations, there won't be any errors at all:
+
+```ts twoslash
+let example1 = "Hello World!";
+let example2 = 42;
+let example3 = true;
+let example4 = Symbol();
+let example5 = 123n;
+```
+
+These basic types are very useful to know, even if you don't always need them for your variable declarations.
+
+#### Solution 4: The `any` Type
+
+In this case, using `any` did not help us at all. In fact, `any` annotations seem to actually turn off type checking!
+
+With the `any` type, we're free to do anything we want to the variable, and TypeScript will not prevent it.
+
+Using `any` also disables useful features like autocompletion, which can help you avoid typos.
+
+That's right-- the error in the above code was caused by a typo of `e.terget` instead of `e.target` when creating the `FormData`!
+
+```ts
+const handleFormData = (e: any) => {
+  e.preventDefault();
+
+  const data = new FormData(e.terget); // e.terget! Whoops!
+
+  const value = Object.fromEntries(data.entries());
+
+  return value;
+};
+```
+
+If `e` had been properly typed, this error would have been caught by TypeScript right away. We'll come back to this example in the future to see the proper typing.
+
+Using `any` may seem like a quick fix when you have trouble figuring out how to properly type something, but it can come back to bite you later.
+
+## Object Literal Types
+
+Now that we've done some exploration with basic types, let's move on to object types.
+
+Object types are used to describe the shape of objects. Each property of an object can have its own type annotation.
+
+When defining an object type, we use curly braces to contain the properties and their types:
+
+```ts twoslash
+const talkToAnimal = (animal: { name: string; type: string; age: number }) => {
+  // rest of function body
+};
+```
```

Comment:
it kinda feels weird for a basic example for object types to be depicted using a function's parameter. I think a type alias would really be more suitable here. Or at least maybe a similar inline annotation on a variable or smth